### DISCUSSION TITLE
Authenticated Requests

### DISCUSSION THREAD

**Discussion by Arinji2** [pocketbase/pocketbase#3598] [Q&A] [answered]
Hiya, so I am not using the normal pb.authstore for auth. Instead using the ssr version of auth. I have the user token with me  in the cookies and the whole auth model like email, id, verified etc. but I dont know how to make authenticated requests to pocketbase. Like since i cant just do pb.authstore.save(token) because its on the server so you dont have access to localstorage.. how do i make a request to pocketbase which satisfies the rules of the collection. Like one of my rules are
@request.auth.id = @collection.playlists.created_by.id

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand what you are doing and what do you mean by _"ssr version of auth"_ but `pb.authstore.save(token)` should work no matter where the SDK instance is created. If `LocalStorage` is not available the default store will keep the token in runtime, aka. until the `pb` object is destroyed.

Also, this seems to be a common misunderstanding and I'll consider either renaming it or document it better, but the `@collection.*` rule should be used only if you want to check a field from other non-directly related collections, aka. to perform a join. If your current collection is "playlists", you can directly use the collection fields without `@collection.playlists.*`. The rule could look something like:
```js
@request.auth.id != "" && @request.auth.id = created_by
```
_(you can remove the first part if `created_by` is marked as "Nonempty")_

**Reply by Arinji2** [reply]
Oh, pb.authstore.save dosent NEED localstorage? Great so i can use that then :D 
Also the created by field is a relation to the users table.

**Reply by ganigeorgiev** [reply]
> Also the created by field is a relation to the users table.

It is OK to be a relation, but:

1. You don't have to perform an explicit join with `@collection.playlists` if "playlists" is the collection where the rule is defined.
2. For **single** relations the end result of `created_by` and `created_by.id` is the same (_`created_by` refers to the same string value that is returned in the record, aka. to the rel id_).
